drivers/counter/counter_mspm0.c

```diff
@@ -42,16 +42,7 @@ static int counter_mspm0_start(const struct device *dev)
 	struct counter_mspm0_data *data = dev->data;
 	const struct counter_mspm0_config *cfg = dev->config;
 	K_SPINLOCK(&data->spinlock) {
-		DL_Timer_reset(cfg->timer);
-		DL_Timer_enablePower(cfg->timer);
-
-		DL_Timer_setClockConfig(cfg->timer, (DL_Timer_ClockConfig *)&cfg->clock_cfg);
 		DL_Timer_initTimerMode(cfg->timer, &data->time_cfg);
```

Comment:
Shouldn't this be done in `counter_mspm0_set_alarm`?

Reply:
you are correct, seems to do the trick